### PR TITLE
Add DMZ Examples, templates 411, 412, and 413

### PR DESCRIPTION
**Pull request by tracsman**
### Contributing guide

https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md
### Changelog
- Adding Examples 1, 2, and 3 from the Azure.com Network Security whitepaper
- Initial upload
### Description of the change

I'm updating the Network Security Best Practice whitepaper on Azure.com to include ARM templates. These are said templates with README.md, Deploy json, parameters json, and metadata json files.

[https://azure.microsoft.com/documentation/articles/best-practices-network-security/](https://azure.microsoft.com/documentation/articles/best-practices-network-security/)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@tracsman__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (jonor). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: 411-dmz-example-1/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,521 @@
+﻿{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "newStorageAccountName": {
```
Comment (singhkay) [inline 411-dmz-example-1/azuredeploy.json]: See #944 for storage guidance

File: 411-dmz-example-1/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+    "imagePublisher": "MicrosoftWindowsServer",
+    "windowsOSVersion": "2012-R2-Datacenter",
+    "vmSize": "Standard_D3",
+    "vmStorageAccountContainerName": "vhds"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 411-dmz-example-1/azuredeploy.json]: Use 2016-01-01 API version for storage. This is a breaking change, see 101-storage-account-create template for usage

File: 411-dmz-example-1/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+      "location": "[resourceGroup().location]",
+      "tags": {
+        "displayName": "StorageAccount"
+      },
+      "properties": {
+        "accountType": "[variables('storageAccountType')]"
+      }
+    },
+    {
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 411-dmz-example-1/azuredeploy.json]: Use GA API version 2015-06-15

File: 411-dmz-example-1/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,521 @@
+﻿{
```
Comment (singhkay) [inline 411-dmz-example-1/azuredeploy.json]: Reviewed first template but comments apply to all

File: 411-dmz-example-1/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Build a DMZ with Network Security Groups (NSG)
```
Comment (singhkay) [inline 411-dmz-example-1/README.md]: A word on the folder naming for these, I would call these something along these lines that talks about the features in the folder name

* 300-dmz-nsg
* 300-dmz-nsg-firewall
* 300-dmz-nsg-firewall-udr

**Comment by gatneil** [comment]
@tracsman any update on this?

File: 411-dmz-example-1/azuredeploy.json (hunk 1 above)
Comment (tracsman) [inline 411-dmz-example-1/azuredeploy.json]: Removed parameter and added a new variable:
"StorageAccountName": "[concat(uniquestring(resourceGroup().id), 'dmz1')]",

**Comment by tracsman** [comment]
Ok, finally getting back to this one. I've removed two templates to make the testing, review, and publish faster. I'll add the other two back in once I complete testing.

Based on feedback above, I've made the following changes:

- Removed storage name parameter and added a new variable:
    "StorageAccountName": "[concat(uniquestring(resourceGroup().id), 'dmz1')]",
- Also removed the parameter from the parameter json file.
- Updated all API versions from preview to current for each object.
- removing all but the first template from this pull request, will fix and re-add the other templates later.
- Renamed the folder from "411-dmz-example-1" to "300-dmz-nsg".
- Updated the metadata.json

I believe this covers all the feedback above.

**Comment by tracsman** [comment]
After three more bug fixes, I think this template is ready to go, @gatneil what do you think?

**Comment by singhkays** [comment]
@tracsman Can you call these templates 301 instead of 300? that makes it inconsistent with all the other templates

**Comment by tracsman** [comment]
@singhkays Updated the folder name.

**Comment by gatneil** [comment]
Could you take a quick look to make sure you're following the best practices checklist? https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

I think I noticed one issue from the checklist, which is the use of 'blob.core.windows.net' for vhd uri.

**Comment by gatneil** [comment]
Any update on this?

**Comment by tracsman** [comment]
@gatneil I've updated the blob references, how's it look now?
Thanks,
-Jon

File: 301-dmz-nsg/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Build a DMZ with Network Security Groups (NSG)
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F300-dmz-nsg%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline 301-dmz-nsg/README.md]: This url should have 301-dmz-nsg, not 300-dmz-nsg

File: 301-dmz-nsg/README.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F300-dmz-nsg%2Fazuredeploy.json" target="_blank">
```
Comment (gatneil) [inline 301-dmz-nsg/README.md]: same

**Comment by gatneil** [comment]
Noticed a small issue with links in the README. Otherwise looks good to go :)

**Comment by gatneil** [comment]
Looks good to me; merging :)

**Comment by tracsman** [comment]
@gatneil Good catch, totally forgot there were references in the ReadMe, they are fixed now.

**Comment by tracsman** [comment]
@gatneil Thanks!!!

**Pull request merged**